src/Core/Services/Implementations/StripePaymentService.cs

```diff
@@ -789,16 +791,25 @@ private async Task ChangeOrganizationSponsorship(Organization org, OrganizationS
                 }
                 catch
                 {
-                    // Need to revert the subscription
-                    await _stripeAdapter.SubscriptionUpdateAsync(sub.Id, new Stripe.SubscriptionUpdateOptions
+                    try
                     {
-                        Items = subscriptionUpdate.RevertItemsOptions(sub),
-                        // This proration behavior prevents a false "credit" from
-                        //  being applied forward to the next month's invoice
-                        ProrationBehavior = "none",
-                        CollectionMethod = collectionMethod,
-                        DaysUntilDue = daysUntilDue,
-                    });
+                        // Need to revert the subscription
+                        await _stripeAdapter.SubscriptionUpdateAsync(sub.Id, new Stripe.SubscriptionUpdateOptions
+                        {
+                            Items = subscriptionUpdate.RevertItemsOptions(sub),
+                            // This proration behavior prevents a false "credit" from
+                            //  being applied forward to the next month's invoice
+                            ProrationBehavior = "none",
+                            CollectionMethod = collectionMethod,
+                            DaysUntilDue = daysUntilDue,
+                        });
+                    }
+                    catch
+                    {
```

Comment:
I don't think I understand the point of having another `try/catch` around this when the `catch` will only re-throw.

Reply:
You are right, the try/catch/throw was redundant. After removing it and doing some more testing. I've added a finally block to ensure the collectionMethod is changed back even if the RevertSub fails. The case in question is a chargeNow customer, but we temporarily change that to send_invoice and have to ensure to change it back.